instant-xml-macros/src/de.rs

```diff
@@ -0,0 +1,289 @@
+                        while let Some(item) = &deserializer.peek_next_tag()? {
+                            match item {
+                                XmlRecord::Open(item) => {
+                                    match get_element(&item.key.as_ref()) {
+                                        #elem_type_match
+                                        __Elements::__Ignore => todo!(),
+                                    }
+
+                                    // Verify prefix
+                                    if let Some(prefix) = &item.prefix {
+                                        // Check if such prefix exist
+                                        if !deserializer.verify_prefix(&prefix) {
+                                            return Err(::instant_xml::Error::UnexpectedPrefix);
+                                        }
+                                        // TODO: Check if prefix is equel to declared prefix
+                                    }
+                                 }
+                                 XmlRecord::Close(tag) => {
+                                    println!("Close: {}", tag);
+                                    if tag == #name {
+                                        break;
+                                    }
+                                },
+                                XmlRecord::Element(_) => panic!("Unexpected element"),
+                            }
+                        }
+
+                        if let Some(attributes_map) = &attributes {
+                            for (key, value) in attributes_map.iter() {
+                                match get_attribute(&key) {
+                                    #attr_type_match
+                                    __Attributes::__Ignore => todo!(),
+                                }
+                            }
+                        }
+
+                        println!("return");
+                        Ok(Self::Value {
+                            #return_val
+                    })
+                    }
+                }
+
+                #namespaces_map;
+                deserializer.deserialize_struct(StructVisitor{}, #name, &namespaces_map)
+            }
+        ))
+        .into();
+
+        Deserializer {
+            input,
+            fn_deserialize,
+            fn_from_xml,
+        }
+    }
+
+    fn is_scalar(value: &str) -> bool {
+        matches!(value, "bool" | "i8" | "i16" | "i32" | "i64" | "u8") // TODO: Fill up
+    }
+
+    fn process_element(
+        field: &syn::Field,
+        index: usize,
+        elements_enum: &mut proc_macro2::TokenStream,
+        elements_names: &mut proc_macro2::TokenStream,
+        elem_type_match: &mut proc_macro2::TokenStream,
+        declare_values: &mut proc_macro2::TokenStream,
+        return_val: &mut proc_macro2::TokenStream,
+    ) {
+        let mut field_name = field.ident.as_ref().unwrap().to_string();
```

Comment:
Looks like there's a lot of duplicated code between `process_element()` and `process_attribute()`.

Reply:
I had it combined and it was supr unreadable, a lot of ifs to follow, it looks similar but it is using different variables names, `attributes` vs `elements`. I would keep it separated at least for now.

Reply:
Hmm, ok.

Reply:
done